### PR TITLE
CompleteNexusOperation request forwarding

### PR DESCRIPTION
**Pull request by pdoerner**
## What changed?

Added support for forwarding `CompleteNexusOperation` requests from standby to active clusters.

## Why?

To avoid breaking operations if namespace failover occurs.

## How did you test it?

New xdc functional test

### CODE REVIEW

File: components/nexusoperations/frontend/handler.go
Comment (bergundy) [inline components/nexusoperations/frontend/handler.go:238]: This will panic if the header is nil although I think it should never be nil (note that `Clone` returns `nil` if the request header was nil).

File: tests/xdc/nexus_request_forwarding_test.go
Comment (bergundy) [inline tests/xdc/nexus_request_forwarding_test.go:646]: Nice, we could use this trick in some other tests.

File: tests/xdc/nexus_request_forwarding_test.go
Comment (bergundy) [inline tests/xdc/nexus_request_forwarding_test.go:454]: You don't need to fail over, just configure the passive cluster's URL as the callback URL.

File: tests/xdc/nexus_request_forwarding_test.go
Comment (bergundy) [inline tests/xdc/nexus_request_forwarding_test.go:465]: I think the TODO addresses this but this step shouldn't be required, if it is, we have a bug.

**Review by bergundy — COMMENTED** [review]
Overall LGTM, let's wait until history apply is merged and merge this.

File: tests/xdc/nexus_request_forwarding_test.go
Comment (pdoerner) [reply tests/xdc/nexus_request_forwarding_test.go:646]: Yeah that's why I exposed the frontend namespace registry in another PR. `DescribeNamespace` goest straight to persistence, so it's not useful for waiting until the cache is updated.

File: tests/xdc/nexus_request_forwarding_test.go
Comment (pdoerner) [reply tests/xdc/nexus_request_forwarding_test.go:454]: I guess I was thinking it would be better to keep it as close to a real scenario as possible. There wouldn't be a reason for a user to set the passive cluster as the callback URL, right? And this has the side benefit of indirectly testing that failover won't break callbacks.

Comment (bergundy) [reply tests/xdc/nexus_request_forwarding_test.go:451]: I wrote a test that covers that already, I'd only do the bare minimum here and verify just the forwarding.

**Review by bergundy — APPROVED** [review]
I only have some minor comments. Overall LGTM.

**Pull request merged**
